src/confcom/azext_confcom/cose_proxy.py

```diff
@@ -0,0 +1,220 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import subprocess
+import os
+import stat
+from pathlib import Path
+import platform
+from zipfile import ZipFile
+import requests
+from azext_confcom.errors import eprint
+from azext_confcom.template_util import pretty_print_func
+from azext_confcom.config import (
+    REGO_CONTAINER_START,
+    REGO_FRAGMENT_START,
+    POLICY_FIELD_CONTAINERS,
+    POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS,
+    POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_ISSUER,
+    POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_FEED,
+    POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_MINIMUM_SVN,
+    ACI_FIELD_CONTAINERS_REGO_FRAGMENTS_INCLUDES,
+)
+
+
+host_os = platform.system()
+machine = platform.machine()
+
+
+class CoseSignToolProxy:  # pylint: disable=too-few-public-methods
+
+    @staticmethod
+    def download_binaries():
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+
+        bin_folder = os.path.join(dir_path, "bin")
+        if not os.path.exists(bin_folder):
+            os.makedirs(bin_folder)
+
+        # get the most recent release artifacts from github
+        r = requests.get("https://api.github.com/repos/microsoft/CoseSignTool/releases")
+        needed_assets = ["CoseSignTool-Windows-release.zip", "CoseSignTool-Linux-release.zip"]
+        windows_flag = False
+        linux_flag = False
+        for release in r.json():
+            # these should be newest to oldest
+            for asset in release["assets"]:
+                # download the file if it's what we want
+                if asset["name"] in needed_assets:
+                    if "Windows" in asset["name"]:
+                        windows_flag = True
+                    else:
+                        linux_flag = True
+
+                    last_dash = asset["name"].rfind("-")
+                    save_name = asset["name"][:last_dash]
+
+                    zip_url = asset["browser_download_url"]
+                    r = requests.get(zip_url)
+                    # save and unzip the file to the bin folder
+                    with open(os.path.join(bin_folder, asset["name"]), "wb") as f:
+                        f.write(r.content)
+                    with ZipFile(os.path.join(bin_folder, asset["name"]), "r") as zip_ref:
+                        zip_ref.extractall(os.path.join(bin_folder, save_name))
+                    # remove the zip file
+                    os.remove(os.path.join(bin_folder, asset["name"]))
+
+            # stop early so we don't have to iterate through all releases
+            if windows_flag and linux_flag:
```

Comment:
Getting all the assets puts them in chronological order so this will always grab the newest release without needing a code change

Reply:
Then shouldn't we stop looking for linux/windows and only look for the other once we have it? This looks like it will continue getting all windows/linux artifacts until it has at least one of both.

Reply:
for the inner for loop, I've had issues in the past with repos changing the names of their executables. This helps that a tiny bit

Reply:
Will update when the needed repo starts publishing its binaries. Until then, this code isn't useful anyways